### DISCUSSION TITLE
OLE object for VBscript etc. is not present anymore

### DISCUSSION THREAD

**Discussion by FrontierDK** [ImageMagick/ImageMagick#4913] [Development]
I just updated my ImageMagick DLL version to the latest version (it's been a while since last updating), and now - the OLE object for Windows scripting host seems to be gone.

Also, it's not an option during install anymore. Is this intentional?

**Comment by urban-warrior** [comment]
It was intentional.  The OLE object has not been maintained for over a decade now and it was considered insecure.  We encourage users to use Magick.NET instead.  You can make a case for the OLE object here if you want.   @dlemstra will review and decide if we need to resurrect the object.

**Comment by FrontierDK** [comment]
Well, it's the one super component to use in both VBscript, classic ASP etc., and thus great for automation. Since it's an optional component (not chosen by default), it shouldn't pose any security problem...

**Comment by FrontierDK** [comment]
For those who use the OLE component, I did some digging around in my archives, and ImageMagick-7.1.0-19 is THE version to go for (it still has the OLE component).

**Reply by AlexLaforge** [reply]
If someone need this **7.1.0-19** version, I managed to keep it as the original Windows Binary as it was downloadable from the IM website and the GitHub releases.
This is very problematic seeing this OLE interface being deprecated.

We absolutely agree that software must evolve, but considering the wide ImageMagick userbase, deprecating this component poses numerous problems and challenges. Being forced to use a **WScript Shell** from VBScript, Classic ASP or other languages is a regression, and makes us more exposed to the numerous CVE's that are regularly discovered, as we are now basically forced to use a "command-line", that is just very unsecure  :-/

Would this resurrection be **this hard** ? I allow myself to ask very kindly to @dlemstra

**Reply by FrontierDK** [reply]
Fortunately, even the latest x86 version still works, if you take ImageMagickObject DLL from version 7.1.0.19 - copy ImageMagickObject.dll to the folder where you installed ImageMagick, register it using regsvr32, and voila - it still works :)

I don't get why some feel it's a good idea to decide for others. Yes, I am aware that it might hold security errors, but I have very strict naming usage anyways.

**Reply by urban-warrior** [reply]
ImageMagick is a free and open-source software, with a (small) team of unpaid volunteers who maintain it. However, given our limited time and resources, we occasionally have to make trade-offs between available time, effort, and security risks. The decision to discontinue support for this particular issue was based on all three of these considerations.

While we regret the impact this may have on our users, we believe it was the best course of action given our current circumstances. However, if an organization were to provide funding, our team could potentially work on ImageMagick full-time, which would allow us to continue supporting the ImageMagickObject DLL.

In the meantime, we want to emphasize that as an open-source software, anyone in the user community can clone the ImageMagickObject from our Github repository and maintain it for the benefit of others. We encourage users who are interested in doing so to contribute to the project and help keep it going.

**Comment by dlemstra** [comment]
We will not restore the OLE component, it is considered deprecated and should not be used anymore.
